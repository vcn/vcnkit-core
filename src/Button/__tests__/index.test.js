import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import 'jest-styled-components';

import * as Button from '../';

describe('Button', () => {
    test('Primary button renders', () => {
        const tree = renderer.create(
            <Button.Primary
                onClick={ () =>{} }
            >
                PrimaryButton
            </Button.Primary>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('color', '#5282B5');
    });
    
    test('Destructive button renders', () => {
        const tree = renderer.create(
            <Button.Destructive
                onClick={ () =>{} }
            >
                DestructiveButton
            </Button.Destructive>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();    
        expect(tree).toHaveStyleRule('color', '#E53935');
    });
    
    test('Calls click-handler when clicked', () => {
        const handleClick = jest.fn();

        const button = mount(
            <Button.Primary
                onClick={ handleClick }
            >
                PrimaryButton
            </Button.Primary>
        );

        button.find('button').getDOMNode().dispatchEvent(new MouseEvent('click'));
        expect(handleClick.mock.calls.length).toEqual(0);
    });

    test('Does not call click-handler when disabled', () => {
        const handleClick = jest.fn();

        const button = mount(
            <Button.Primary
                onClick={ handleClick }
                disabled
            >
                PrimaryButton
            </Button.Primary>
        );

        button.find('button').getDOMNode().dispatchEvent(new MouseEvent('click'));
        expect(handleClick.mock.calls.length).toEqual(0);
    });

    test('Renders a link', () => {
        const href = 'https://www.google.com';
        const target = '_blank';
        const rel = 'nofollow';
    
        const button = mount(
            <Button.Primary
                href={ href }
                target={ target }
                rel={ rel }
            >
                PrimaryButton
            </Button.Primary>
        );
    
        const domNode = button.getDOMNode();
    
        expect(domNode.tagName).toBe('A');
        expect(domNode.getAttribute('href')).toBe(href);
        expect(domNode.getAttribute('target')).toBe(target);
        expect(domNode.getAttribute('rel')).toBe(rel);
    });
});
