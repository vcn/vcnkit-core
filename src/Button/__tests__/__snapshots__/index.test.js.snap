// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Destructive button renders 1`] = `
.c0 {
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: .8125rem;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  white-space: nowrap;
  padding: 0 .5rem;
  min-width: 4rem;
  min-height: 2rem;
  border: none;
  border-radius: .125rem;
  font-family: Roboto,sans-serif;
  color: hsla(0,0%,0%,.87);
  color: #E53935;
}

.c0::-moz-focus-inner {
  border: none;
}

.c0:focus {
  outline: none;
}

.c0:disabled,
.c0[aria-disabled] {
  color: hsla(0,0%,0%,.12);
  cursor: not-allowed;
}

.c0:not(:disabled):not([aria-disabled]) {
  cursor: pointer;
}

.c0:not(:disabled):not([aria-disabled]):hover,
.c0:not(:disabled):not([aria-disabled]):focus {
  background: hsla(0,0%,0%,.06);
}

<button
  className="c0"
  onClick={[Function]}
  type="button"
>
  DestructiveButton
</button>
`;

exports[`Button Primary button renders 1`] = `
.c0 {
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: .8125rem;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  white-space: nowrap;
  padding: 0 .5rem;
  min-width: 4rem;
  min-height: 2rem;
  border: none;
  border-radius: .125rem;
  font-family: Roboto,sans-serif;
  color: hsla(0,0%,0%,.87);
  color: #5282B5;
}

.c0::-moz-focus-inner {
  border: none;
}

.c0:focus {
  outline: none;
}

.c0:disabled,
.c0[aria-disabled] {
  color: hsla(0,0%,0%,.12);
  cursor: not-allowed;
}

.c0:not(:disabled):not([aria-disabled]) {
  cursor: pointer;
}

.c0:not(:disabled):not([aria-disabled]):hover,
.c0:not(:disabled):not([aria-disabled]):focus {
  background: hsla(0,0%,0%,.06);
}

<button
  className="c0"
  onClick={[Function]}
  type="button"
>
  PrimaryButton
</button>
`;
